R/sort.R

```diff
@@ -0,0 +1,38 @@
+#' Automatically sort factor levels according to a user-defined function
+#'
+#' @param .f A factor (or character vector).
+#' @param .fun A function that will sort or permute the existing factor levels.
+#'   It must accept one character argument and return a character argument of
+#'   the same length as it's input.
+#'
+#' @param ... Additional arguments to `.fun`.
+#' @export
+#' @examples
+#' chromosomes <- c("chr2", "chr1", "chr10")
+#' chr_fac <- factor(chromosomes, levels = chromosomes)
+#' chr_fac
+#'
+#' # naive alphanumeric sorting "1" < "10" < "2"
+#' fct_sort(chr_fac, sort)
+#'
+#'  # number-based alphanumeric sorting "1" < "2" < "10"
```

Comment:
I think this is maybe a bit complicated for an example? Maybe just do something with alphabetical sorting? And maybe `sample()` to show random reordering?

Reply:
I replaced the example. The new examples includes

- alphabetical-sort,
- alphabetical-decreasing-sort (equiv to fct_rev)
- alphabetical sort with an out-of-order baseline level (equiv to fct_relevel)
- sampling from the levels